src/common.speech/SynthesisAdapterBase.ts

```diff
@@ -0,0 +1,630 @@
+        });
+    }
+
+    public static addHeader(audio: ArrayBuffer, format: AudioOutputFormatImpl): ArrayBuffer {
+        if (!format.hasHeader) {
+            return audio;
+        }
+        format.updateHeader(audio.byteLength);
+        const tmp = new Uint8Array(audio.byteLength + format.header.byteLength);
+        tmp.set(new Uint8Array(format.header), 0);
+        tmp.set(new Uint8Array(audio), format.header.byteLength);
+        return tmp.buffer;
+    }
+
+    public isDisposed(): boolean {
+        return this.privIsDisposed;
+    }
+
+    public dispose(reason?: string): void {
+        this.privIsDisposed = true;
+        if (this.privSessionAudioDestination !== undefined) {
+            this.privSessionAudioDestination.close();
+        }
+        if (this.privConnectionConfigurationPromise) {
+            this.privConnectionConfigurationPromise.onSuccessContinueWith((connection: IConnection) => {
+                connection.dispose(reason);
+            });
+        }
+    }
+
+    public connect(): void {
+        this.connectImpl().result();
+    }
+
+    public connectAsync(cb?: Callback, err?: Callback): void {
+        this.connectImpl().continueWith((promiseResult: PromiseResult<IConnection>) => {
+            try {
+                if (promiseResult.isError) {
+                    if (!!err) {
+                        err(promiseResult.error);
+                    }
+                } else if (promiseResult.isCompleted) {
+                    if (!!cb) {
+                        cb();
+                    }
+                }
+            } catch (e) {
+                if (!!err) {
+                    err(e);
+                }
+            }
+        });
+    }
+
+    public Speak(
+        text: string,
+        isSSML: boolean,
+        requestId: string,
+        successCallback: (e: SpeechSynthesisResult) => void,
+        errorCallBack: (e: string) => void,
+        audioDestination: IAudioDestination,
+    ): Promise<boolean> {
+
+        let ssml: string;
+
+        if (isSSML) {
+            ssml = text;
+        } else {
+            ssml = SpeechSynthesizer.buildSsml(text, this.privSynthesizerConfig.parameters);
+        }
+
+        if (this.speakOverride !== undefined) {
+            return this.speakOverride(ssml, requestId, successCallback, errorCallBack);
+        }
+
+        this.privSuccessCallback = successCallback;
+        this.privErrorCallback = errorCallBack;
+
+        this.privSynthesisTurn.startNewSynthesis(requestId, text, isSSML, audioDestination);
+
+        // Start the connection to the service. The promise this will create is stored and will be used by configureConnection().
+        this.connectImpl();
+
+        return this.fetchConnection().continueWithPromise<boolean>((connection: PromiseResult<IConnection>) => {
+            if (connection.isError) {
+                this.cancelSynthesisLocal(CancellationReason.Error, CancellationErrorCode.ConnectionFailure, connection.error);
```

Comment:
You need to return here also, right? Else it'll try to send the context on the error'd connection

Reply:
Yes, for sure it'll send the context. Thanks for pointing this, will fix.